src/builder_state.rs

```diff
@@ -404,29 +439,38 @@ impl<TYPES: NodeType> BuilderProgress<TYPES> for BuilderState<TYPES> {
         // if you haven't launched the clone, then you don't exit, you need atleast one clone to function properly
         // the special value can be 0 itself, or a view number 0 is also right answer
         let leaf_chain = decide_msg.leaf_chain;
-        let _qc = decide_msg.qc;
+        //let _qc = decide_msg.qc;
         let _block_size = decide_msg.block_size;
 
-        let _latest_decide_parent_commitment = leaf_chain[0].leaf.parent_commitment;
+        let _latest_decide_parent_commitment = leaf_chain[0].leaf.get_parent_commitment();
 
         let _latest_decide_commitment = leaf_chain[0].leaf.commit();
 
-        let leaf_view_number = leaf_chain[0].leaf.view_number;
+        let latest_leaf_view_number = leaf_chain[0].leaf.get_view_number();
 
         // bootstrapping case
         // handle the case when we hear a decide event before we have atleast one clone, in that case, we might exit the builder
         // and not make any progress; so we need to handle that case
         // Adhoc logic: if the number of subscrived receivers are more than 1, it means that there exists a clone and we can safely exit
-        if self.built_from_view_vid_leaf.0.get_u64() == 0 && self.tx_receiver.receiver_count() <= 1
-        {
-            tracing::info!("In bootstrapping phase");
+        if self.built_from_view_vid_leaf.0.get_u64() == self.bootstrap_view_number.get_u64() {
+            tracing::info!("Bootstrapped builder state, should continue");
+            // split_off returns greater than equal to set, so we want everything after the latest decide event
+            let splitted_list = self
```

Comment:
`slitted_list` is grammatically awkward. `split_list` instead. I know, that's also the imperative form, but English is a messy language.

Reply:
https://github.com/EspressoSystems/hs-builder-core/pull/68/commits/832462931d7160e8b11e661e9859098fd5f20cc3